django_deprecate_fields/deprecate_field.py

```diff
@@ -32,7 +32,15 @@ def __get__(self, obj, type=None):
         logger.warning(msg)
         if not callable(self.val):
             return self.val
-        return self.val()
+        try:
+            # we pass the Model object instance to the callable in case the
+            # value of the deprecated field can be extrapolated from another
+            # field
+            return self.val(obj)
+        except TypeError:
+            # this is a compatibility fallback in case the callback does not
+            # support any arguments
+            return self.val()
```

Comment:
This is lazy but should be fine in most cases, otherwise I can implement it with `inspect.getfullargspec()`